saleor/account/views.py

```diff
@@ -75,17 +75,32 @@ def password_reset_confirm(request, uidb64=None, token=None):
 
 @login_required
 def details(request):
-    password_form = get_or_process_password_form(request)
+
+    email_form, password_form = get_forms_by_method_or_button(request)
     orders = request.user.orders.confirmed().prefetch_related('lines')
     orders_paginated = get_paginator_items(
         orders, settings.PAGINATE_BY, request.GET.get('page'))
     ctx = {'addresses': request.user.addresses.all(),
            'orders': orders_paginated,
-           'change_password_form': password_form}
+           'change_password_form': password_form,
+           'change_email_form': email_form}
 
     return TemplateResponse(request, 'account/details.html', ctx)
 
 
+def get_forms_by_method_or_button(request):
+    if request.method == 'POST':
+        if 'email_change' in request.POST:
+            password_form = ChangePasswordForm(data=None, user=request.user)
+            email_form = email_edit(request)
+        elif 'password_change' in request.POST:
+            password_form = get_or_process_password_form(request)
+            email_form = EmailChangeForm(data=None)
+        return email_form, password_form
+    else:
```

Comment:
This `else` statement is not necessary, as you'll return from the above's `if` anyway